src/xenia/base/clock.cc

```diff
@@ -68,43 +132,58 @@ void Clock::set_guest_system_time_base(uint64_t time_base) {
 }
 
 uint64_t Clock::QueryGuestTickCount() {
-  UpdateGuestClock();
-  return guest_tick_count_;
+  auto guest_tick_count = UpdateGuestClock();
+  return guest_tick_count;
 }
 
 uint64_t Clock::QueryGuestSystemTime() {
-  UpdateGuestClock();
-  return guest_system_time_base_ + guest_time_filetime_;
-}
+  if (cvars::clock_no_scaling) {
+    return Clock::QueryHostSystemTime();
+  }
 
-uint32_t Clock::QueryGuestUptimeMillis() {
-  UpdateGuestClock();
-  uint64_t uptime_millis = guest_tick_count_ / (guest_tick_frequency_ / 1000);
-  uint32_t result = uint32_t(std::min(uptime_millis, uint64_t(UINT_MAX)));
-  return result;
+  auto guest_system_time_offset = QueryGuestSystemTimeOffset();
+  return guest_system_time_base_ + guest_system_time_offset;
 }
 
-void Clock::SetGuestTickCount(uint64_t tick_count) {
-  last_host_tick_count_ = Clock::QueryHostTickCount();
-  guest_tick_count_ = tick_count;
+uint32_t Clock::QueryGuestUptimeMillis() {
+  return static_cast<uint32_t>(
+      std::min(QueryGuestSystemTimeOffset() / 10000,
+               static_cast<uint64_t>(std::numeric_limits<uint32_t>::max())));
```

Comment:
Cast to `uint64_t` can probably be removed by specializing `std::min<uint64_t>` here?